Add unixy option to display only 'changed' output

akatch opened the pull request:
##### SUMMARY
This PR adds an option that, when set, will display task output only for
tasks that return a 'changed' (or failed) result.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
callback plugin unixy.py

##### ANSIBLE VERSION
```
ansible 2.8.0.dev0 (unixy_show_only_changed d87e63cfe7) last updated 2018/10/09 23:41:39 (GMT -500)
  config file = /home/ally/.ansible.cfg
  configured module search path = [u'/home/ally/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ally/code/ansible/lib/ansible
  executable location = /home/ally/code/ansible/bin/ansible
  python version = 2.7.13 (default, Jan 19 2017, 14:48:08) [GCC 6.3.0 20170118]
```

##### ADDITIONAL INFORMATION

```
# Default behavior
Executing playbook test.yml

- localhost -
Gathering Facts...
  localhost ok
debug...
  localhost ok: {
    "changed": false,
    "msg": "Success."
} | msg: Success.
debug...
  localhost ok: {
    "ansible_hostname": "metis",
    "changed": false
}
command...
  localhost done | stdout:  23:35:51 up 23 days,  1:05, 27 users,  load average: 0.25, 0.11, 0.13

- Play recap -
  localhost                  : ok=4    changed=1    unreachable=0    failed=0

# with UNIXY_SHOW_ONLY_CHANGED=true
Executing playbook test.yml

- localhost -
Gathering Facts...
debug...
debug...
command...
  localhost done | stdout:  23:35:41 up 23 days,  1:05, 27 users,  load average: 0.20, 0.10, 0.13

- Play recap -
  localhost                  : ok=4    changed=1    unreachable=0    failed=0

```

Commit:
Add unixy option to display only 'changed' output

This PR adds an option that, when set, will display task output only for
tasks that return a 'changed' (or failed) result.

Commit:
Add unixy option to display only 'changed' output

This PR adds an option that, when set, will display task output only for
tasks that return a 'changed' (or failed) result.

Commit:
Clarify documentation

Commit:
Clarify documentation

jamescassell commented on lib/ansible/plugins/callback/unixy.py:
```diff
@@ -18,6 +18,15 @@
+        description: If true, do not show output from 'ok' task results
+        default: False
+        env:
+          - name: UNIXY_SHOW_ONLY_CHANGED
```
If you're defining a new environment variable, it should be prefixed with`ANSIBLE_`

akatch replied:
I pulled this out, but will keep in mind for next time. Thanks for the feedback.

Commit:
Implement display_ok_hosts and display_skipped_hosts like default callback

Commit:
Implement display_ok_hosts and display_skipped_hosts like default callback

Commit:
Implement show_custom_stats and display_failed_stderr like default callback

Commit:
Implement show_custom_stats and display_failed_stderr like default callback

acdha committed:
docs: SAN prefixes are required for openssl_csr (#53144)

The openssl_csr docs use “should” where (at least following IETF nomenclature) “must” would be more accurate.

See #53141 

+label: docsite_pr

yungezz committed:
fix keyvault auth in msi auth (#53100)

yungezz committed:
fix class name (#53162)

RedWhiteMiko committed:
Do not delete duplicate host when fetching Zabbix Hosts (#53160)

fanf2 committed:
get_url: return no change in check mode when checksum matches (#53070)

Signed-off-by: Tony Finch <dot@dotat.at>

machacekondra committed:
kubevirt: Fix cloud_init in kubevirt_vm (#52999)

mmazur committed:
Add kubevirt_cdi_upload module (#52990)

xmadsen committed:
Resolved conflicts with mistaken commits in earlier history (#52953)

xmadsen committed:
Remove mistaken commits from other pull request. (#52928)

migumun committed:
Fortinet's FortiOS wireless controller wtp (#52850)

* Fortinet's FortiOS wireless controller wtp

* Avoid using global

migumun committed:
Fortinet's FortiOS wireless controller wids profile (#52849)

* Fortinet's FortiOS wireless controller wids profile

* Avoid using global

migumun committed:
Fortinet's FortiOS wireless controller vap (#52847)

* Fortinet's FortiOS wireless controller vap

* Avoid using global

migumun committed:
Fortinet's FortiOS wireless controller utm profile (#52846)

* Fortinet's FortiOS wireless controller utm profile

* Avoid using global

migumun committed:
Fortinet's FortiOS wireless controller setting (#52845)

* Fortinet's FortiOS wireless controller setting

* Avoid using global

felixfontein committed:
Don't die when kernel memory accounting is not supported by runc build. (#53163)

rubentsirunyan committed:
Fixing mediatype default value (#53114)

* Fixing mediatype default value

* Updating the documentation of media_type

rubentsirunyan committed:
[zabbix_action] Refactoring the argument spec (#53106)

* Refactoring the argument spec

* Adding subarguments for recovery and acknowledge operations

* Correcting the linting errors

mkrizek committed:
profile_tasks: Fix last task time with multiple plays (#53065)

Commit:
Merge branch 'unixy_show_only_changed' of github.com:/akatch/ansible into unixy_show_only_changed

Commit:
Fix pep8

akatch commented:
I did a bad job so I'm going to reopen this with a new branch.
